src/main/haskell/language-kore/src/Data/Kore/AST/Common.hs

```diff
@@ -39,6 +39,22 @@ import           Data.Proxy
 
 import           Data.Kore.AST.MetaOrObject
 
+data FileLocation = FileLocation
+    { fileName :: FilePath
+    , line     :: Int
+    , column   :: Int
+    }
+    deriving Show
```

Comment:
If I may. In K, we have two classes that represent location information for terms: Location(<startLine>, <startColumn>, <endLine>, <endColumn>), Source(<string>).
When I'm trying to parse a term, sometimes I get ambiguities, and knowing where parsing ended can be really useful. The source of a term may be coming from other locations than files alone. Command line arguments, generated by front-end desugar steps... I've also been talking with @grosu lately and I think a more appropriate location format would be to have 6 fields: Location(<startLine>, <startColumn>, <endLine>, <endColumn>, <offset>, <length>). This would cover a wider range of situations and allow us to do more things with those terms. (one example that came up was to check if you have whitespace before or after a specific term)

Reply:
Thanks, that's useful. I intended the FileLocation to be an actual position in a file, and the AstLocation to be the location for an AST term, and it has a much richer representation. However, I didn't include file ranges, which are likely to be useful.

I may or may not change its representation right now, but even if I don't, I added a comment explaining that it should be treated as an opaque token as much as possible, so it should be easy to change.